crates/fuel-core/src/database.rs

```diff
@@ -194,7 +194,7 @@ impl Database {
     #[cfg(feature = "rocksdb")]
     pub fn open(path: &Path, capacity: impl Into<Option<usize>>) -> DatabaseResult<Self> {
         use anyhow::Context;
-        let db = RocksDb::default_open(path, capacity.into()).context("Failed to open rocksdb, you may need to wipe a pre-existing incompatible db `rm -rf ~/.fuel/db`")?;
+        let db = RocksDb::default_open(path, capacity.into()).map_err(anyhow::Error::msg).context("Failed to open rocksdb, you may need to wipe a pre-existing incompatible db `rm -rf ~/.fuel/db`")?;
```

Comment:
You can implement `From<DatabaseError>`

Reply:
Yes. I have now implemented `From` for the error classes. In cases where we use `context`, we still need to map to `anyhow::Error` because it is not implemented automatically for non-std errors.